src/goto-programs/slice_global_inits.cpp

```diff
@@ -44,26 +41,8 @@ void slice_global_inits(goto_modelt &goto_model)
   if(e_it==goto_functions.function_map.end())
     throw "entry point not found";
 
-  worklist.push_back(entry_point);
-
-  do
-  {
-    const irep_idt id=worklist.front();
-    worklist.pop_front();
-
-    functions_reached.insert(id);
-
-    const auto &p=graph.equal_range(id);
-
-    for(auto it=p.first; it!=p.second; it++)
-    {
-      const irep_idt callee=it->second;
-
-      if(functions_reached.find(callee)==functions_reached.end())
-        worklist.push_back(callee);
-    }
-  }
-  while(!worklist.empty());
+  std::unordered_set<irep_idt, irep_id_hash> functions_reached=
+      call_graph.reachable_functions(entry_point);
 
```

Comment:
Are there other uses of the callgraph like this?  It seems finding reachable functions might be a common thing to do.

Reply:
As far as I can find, this is the only bit of code that uses this using a graph.

show_call_sequences outputs the function calls, but does it on the fly without constructing the graph. 

find_used_functions obviously needs to compute the reachable functions, but again does it on the fly without constructing a graph.

Reply:
OK.  I guess I was suggesting that it should become an API and maybe in time these / others could be encouraged to use it.